hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto

```diff
@@ -1799,10 +1800,22 @@ message SnapshotDiffResponse {
     DONE = 3;
     REJECTED = 4;
     FAILED = 5;
+    CANCELED = 6;
   }
+
+  enum CancelStatusProto {
```

Comment:
IMO this is more of a `Result` to the one-time client request rather than a `Status` for the diff job.

```suggestion
  enum JobCancelResultProto {
```

e.g. https://github.com/apache/ozone/blob/bdd26eeca9fe7adfd87a4ee1e0acfd84d6148c57/hadoop-hdds/interface-server/src/main/proto/ScmServerProtocol.proto#L182-L199